html/docs/design/sync-ot.html

```diff
@@ -0,0 +1,435 @@
+                </p>
+              </li>
+              <li>
+                <p>
+                Data loss on a server would prevent replicas from synchronizing.
+                It may be possible for the server to produce "backups" that can be stored with replicas and used to reproduce a server state from the past when necessary.
+                </p>
+              </li>
+            </ul>
+            </p>
+
+            <h4 id="servers">Singleton Replicas</h4>
+
+            <p>
+            A replica depends on periodic synchronization for performant operation.
+            Without synchronization, its list of pending operations would grow indefinitely, and tasks could never be expired.
+            So all replicas, even "singleton" replicas which do not replicate task data with any other replica, must synchronize periodically.
+            To support this condition, Taskwarrior provides a local implementation of this model against which sync operations can be performed, and a migration path from such local sync to remote synchronization.
+            </p>
+
+            <a name="model"></a>
+            <h3>Implementation</h3>
+
+            <p>
+            Achieving this approach in Taskwarrior requires some difficult changes.
+            </p>
+
+            <ul>
+              <li>
+                <p>
+                All modifications to tasks must be made in the form of operations.
+                This is not conceptually difficult, but represents a significant refactor of the existing code.
+                The most challenging aspect of this change has <a href="https://github.com/GothenburgBitFactory/taskwarrior/issues/2553">already</a> <a href="https://github.com/GothenburgBitFactory/taskwarrior/issues/2569">been completed</a>: changes such as adding or removing a tag must be formulated as single operations that can be easily transposed.
+                For example, consider conflicting operations <code>Update($uuid, "tags", "foo,bar,bing")</code> and <code>Update($uuid, "tags", "foo,bar,next")</code>.
+                Without an understanding of tags, transposing these two operations requires one to "win", effectively undoing the other.
+                If, instead, each tag is stored as a distinct task key, then the two operations to add tags are <code>Update($uuid, "tag_bing", "")</code> and <code>Update($uuid, "tag_next", "")</code>.
+                These operations do not conflict with each other and can be transposed easily.
+                </p>
+              </li>
+
+              <li>
+                <p>
+                Synchronization becomes a required kind of periodic maintenance, even if against a local implementation.
+                This may be performed automatically, or existing users may be required to run <code>task sync</code> periodically.
```

Comment:
> Synchronization becomes a required kind of periodic maintenance, even if against a local implementation. 
                 
It's a bit unclear what this paragraph is trying to say. Would users that don't use `sync` be required to run some dummy version of `sync`? I don't see evidence for that in the rest of the document, but I'm not sure how to interpret this.

Reply:
That is an option, but we might also be able to do so automatically for cases where synchronization is not configured.  I'm trying to leave that open-ended here.